src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -4443,5 +4462,695 @@ public void GetSimpleNonTypeMembers_DirectApiCheck()
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Error").WithArguments("Error").WithLocation(4, 13)
+                );
+        }
+
+        [Fact]
+        public void ToString_UserDefinedPrintMembers_WrongReturnType()
+        {
+            var src = @"
+record struct C1
+{
+    private int PrintMembers(System.Text.StringBuilder builder) => throw null;
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (4,17): error CS8874: Record member 'C1.PrintMembers(StringBuilder)' must return 'bool'.
+                //     private int PrintMembers(System.Text.StringBuilder builder) => throw null;
+                Diagnostic(ErrorCode.ERR_SignatureMismatchInRecord, "PrintMembers").WithArguments("C1.PrintMembers(System.Text.StringBuilder)", "bool").WithLocation(4, 17)
+                );
+        }
+
+        [Fact]
+        public void ToString_UserDefinedPrintMembers()
+        {
+            var src = @"
+var c1 = new C1();
+System.Console.Write(c1.ToString());
+
+record struct C1
+{
+    private bool PrintMembers(System.Text.StringBuilder builder)
+    {
+        builder.Append(""RAN"");
+        return true;
+    }
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "C1 { RAN }");
+        }
+
+        [Fact]
+        public void ToString_UserDefinedPrintMembers_WrongAccessibility()
+        {
+            var src = @"
+var c = new C1();
+System.Console.Write(c.ToString());
+
+record struct C1
+{
+    internal bool PrintMembers(System.Text.StringBuilder builder) => throw null;
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (7,19): error CS8879: Record member 'C1.PrintMembers(StringBuilder)' must be private.
+                //     internal bool PrintMembers(System.Text.StringBuilder builder) => throw null;
+                Diagnostic(ErrorCode.ERR_NonPrivateAPIInRecord, "PrintMembers").WithArguments("C1.PrintMembers(System.Text.StringBuilder)").WithLocation(7, 19)
+                );
+        }
+
+        [Fact]
+        public void ToString_UserDefinedPrintMembers_Static()
+        {
+            var src = @"
+record struct C1
+{
+    static private bool PrintMembers(System.Text.StringBuilder builder) => throw null;
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (4,25): error CS8877: Record member 'C1.PrintMembers(StringBuilder)' may not be static.
+                //     static private bool PrintMembers(System.Text.StringBuilder builder) => throw null;
+                Diagnostic(ErrorCode.ERR_StaticAPIInRecord, "PrintMembers").WithArguments("C1.PrintMembers(System.Text.StringBuilder)").WithLocation(4, 25)
+                );
+        }
```

Comment:
>} [](start = 8, length = 1)

Are we testing a scenario when synthesized ```PrintMembers``` used explicitly by user code?

Reply:
>> Are we testing a scenario when synthesized PrintMembers used explicitly by user code?

> Updated ToString_UserDefinedPrintMembers to cover

It looks like it doesn't call "synthesized PrintMembers".

---
In reply to: [601913646](https://github.com/dotnet/roslyn/pull/52012#discussion_r601913646) [](ancestors = 601913646,601724459)

Reply:
Added `ToString_CallingSynthesizedPrintMembers`

---
In reply to: [601929272](https://github.com/dotnet/roslyn/pull/52012#discussion_r601929272) [](ancestors = 601929272,601913646,601724459)